src/TraceManager.py

```diff
@@ -0,0 +1,129 @@
+"""
+<Program Name>
+  TraceManager
```

Comment:
Ok I officially don't like the name TraceManager for this entity anymore.  I think its better called BacklogManager or something like that.